rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -1092,7 +1131,11 @@ void PlayerImpl::prepare_publishers()
   }
 
   // Create topic publishers
-  auto topics = reader_->get_all_topics_and_types();
+  std::vector<rosbag2_storage::TopicMetadata> topics{};
+  for (const auto & [reader, _] : input_bags_) {
+    auto bag_topics = reader->get_all_topics_and_types();
+    topics.insert(topics.begin(), bag_topics.begin(), bag_topics.end());
```

Comment:
Inserting to the end may save us from an extra copy of other elements.
```suggestion
    topics.insert(topics.end(), bag_topics.begin(), bag_topics.end());
```

Reply:
fixed